### ISSUE TITLE
Unable to Delete Incidents via API

### ISSUE THREAD

**Issue by mrthankyou** [Netflix/dispatch#716]
**Describe the bug**
I am unable to delete incidents via Dispatch's API. 

**To Reproduce**
1. Create an incident
2. Submit the following HTTP request

```
DELETE /api/v1/incidents/<INCIDENT_ID_HERE> HTTP/1.1
Host: localhost:8000
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:82.0) Gecko/20100101 Firefox/82.0
Accept: application/json, text/plain, */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Authorization: Bearer <API_TOKEN>
Referer: http://localhost:8000/incidents
Connection: close
Cache-Control: max-age=0
Content-Type: application/x-www-form-urlencoded
Content-Length: 0
```

3. Receive the following HTTP request

```
HTTP/1.1 500 Internal Server Error
date: Mon, 14 Dec 2020 23:09:01 GMT
server: uvicorn
content-length: 21
content-type: text/plain; charset=utf-8
strict-transport-security: max-age=31536000 ; includeSubDomains
connection: close

Internal Server Error
```
 

**Expected behavior**
The incident should be deleted

**Screenshots**
N/A

**Desktop (please complete the following information):**
Mac OS 10.14.6

**Smartphone (please complete the following information):**
N/A

**Additional context**
When reviewing the Dispatch logs, I see the following error for when I made the DELETE request:

```
web_1        | sqlalchemy.exc.IntegrityError: (psycopg2.errors.ForeignKeyViolation) update or delete on table "incident" violates foreign key constraint "event_incident_id_fkey" on table "event"
web_1        | DETAIL:  Key (id)=(19) is still referenced from table "event".
```

**Comment by mvilanova** [comment]
@mrthankyou we haven't had time to define the cascade behavior for relationships in the data models yet, therefore deleting incidents or other data objects that have parent-child relationships is currently not supported. We're hoping we have free cycles to define them at the beginning of next year. Sorry for the inconvenience this may be causing.

**Comment by mrthankyou** [comment]
Thanks for the quick response! No inconvenience at all.

**Comment by mvilanova** [comment]
We can leave it open for others to see and close it once we merge the necessary changes to master.

**Comment by mvilanova** [comment]
Started working on it [here](https://github.com/Netflix/dispatch/pull/727) if you want to follow. It needs more work and testing before we can merge it.

**Comment by mvilanova** [comment]
@mrthankyou I merged #727 and #748 which allow admins to delete incidents via Web UI and API. The next release will include the changes. Let me know if you find any issues with them.

**Comment by mrthankyou** [comment]
Thanks @mvilanova!
